app/src/test/java/com/weatherxm/data/services/InMemoryTest.kt

```diff
@@ -0,0 +1,146 @@
+package com.weatherxm.data.services
+
+import arrow.core.Either
+import com.mapbox.search.result.SearchSuggestion
+import com.weatherxm.TestConfig.sharedPref
+import com.weatherxm.TestUtils.isSuccess
+import com.weatherxm.data.models.CountryInfo
+import com.weatherxm.data.models.DataError
+import com.weatherxm.data.models.Failure
+import com.weatherxm.data.models.Location
+import com.weatherxm.data.models.User
+import com.weatherxm.data.services.CacheService.Companion.KEY_DEVICES_OWN
+import com.weatherxm.data.services.CacheService.Companion.KEY_HAS_WALLET
+import com.weatherxm.data.services.CacheService.Companion.KEY_USER_ID
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.types.shouldBeTypeOf
+import io.mockk.every
+import io.mockk.mockk
+
+class InMemoryTest(private val cacheService: CacheService) {
+    private val walletAddress = "walletAddress"
+    private val user = User("id", "email@email.com", null, null, null, null)
+    private val searchSuggestion = mockk<SearchSuggestion>().apply {
+        every { id } returns "searchSuggestionId"
+    }
+    private val searchSuggestions = listOf(searchSuggestion)
+    private val location = Location(0.0, 0.0)
+    private val query = "query"
+    private val deviceId = "deviceId"
+    private val deviceIds = listOf(deviceId)
+    private val countriesInfo = listOf<CountryInfo>(mockk())
+
+    private fun BehaviorSpec.testInMemoryEither(
+        dataTitle: String,
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/353#issuecomment-2392359926>#d0cda9</a><span> - 10/03/2024, 09:38 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The testInMemoryEither function modifies the state of the CacheService, which could lead to interdependent tests. A similar issue was also found in app/src/test/java/com/weatherxm/data/services/InMemoryTest.kt (line 48).
</div>
<div id="fix">
<b>Fix:</b> Refactor the testInMemoryEither function to reset the state of the CacheService before each test. This can be done by adding a setup method that clears the cache before each test run.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -34,6 +34,7 @@
      ): BehaviorSpec {
          context("GET / SET $dataTitle") {
 +            beforeTest { cacheService.clear() }
              When("GET the $dataTitle") {
                  and("The $dataTitle is null") {
                      then("return CacheMissError") {
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged